tests/components/aosmith/test_device.py

```diff
@@ -0,0 +1,24 @@
+"""Tests for the device created by the A. O. Smith integration."""
+
+from homeassistant.components.aosmith.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from tests.common import MockConfigEntry
+
+
+async def test_device_nickname(
+    hass: HomeAssistant, init_integration: MockConfigEntry
+) -> None:
+    """Test creation of the device."""
+    device_registry = dr.async_get(hass)
+    reg_device = device_registry.async_get_device(
+        identifiers={(DOMAIN, "junctionId")},
+    )
+
+    assert reg_device.manufacturer == "A. O. Smith"
```

Comment:
Maybe look into snapshot testing, that would make this test way cleaner

Reply:
Updated.